### PR TITLE
Add all CAPI CRDs

### PR DESCRIPTION
**Pull request by MarcelMue**
WE DON'T NEED CHECKLISTS WHERE WE ARE GOING

towards https://github.com/giantswarm/giantswarm/issues/15997

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add CAPI CRDs : `AzureManagedCluster`, `AzureManagedControlPlane`, `AzureManagedMachinePool`, `AzureMachineTemplate`, `Machine`, `MachineSet`, `MachineHealthCheck`, `KubeadmConfig`, `KubeadmConfigTemplate`, `KubeadmControlPlane`.
```
Comment (QuentinBisson) [inline CHANGELOG.md:12]: I feel like we are missing a bunch here :D

Comment (MarcelMue) [reply CHANGELOG.md:12]: I didn't keep adding them here :D

File: hack/generate-manifests.sh [hunk 2]
```diff
@@ -42,15 +42,33 @@ for version in v1 v1beta1; do
...
-  rm ../config/crd/$version/cluster.x-k8s.io_machinesets.yaml
-  rm ../config/crd/$version/cluster.x-k8s.io_machinehealthchecks.yaml
-  rm ../config/crd/$version/infrastructure.cluster.x-k8s.io_azuremachinetemplates.yaml
-  rm ../config/crd/$version/exp.infrastructure.cluster.x-k8s.io_azuremanagedclusters.yaml
-  rm ../config/crd/$version/exp.infrastructure.cluster.x-k8s.io_azuremanagedcontrolplanes.yaml
-  rm ../config/crd/$version/exp.infrastructure.cluster.x-k8s.io_azuremanagedmachinepools.yaml
+  # With Kubeadm related types.
+  ./tools/bin/controller-gen \
+    crd \
+    paths=sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3 \
```
Comment (MarcelMue) [inline hack/generate-manifests.sh:48]: @QuentinBisson We need a path like this for the csi CRD

File: hack/capi.go [hunk 3]
```diff
@@ -4,5 +4,7 @@ package hack
 
 import (
 	_ "sigs.k8s.io/cluster-api"
+	_ "sigs.k8s.io/cluster-api-provider-aws"
 	_ "sigs.k8s.io/cluster-api-provider-azure"
+	_ "sigs.k8s.io/cluster-api-provider-vsphere"
```
Comment (MarcelMue) [inline hack/capi.go:9]: And here we also need to add the csi thingy @QuentinBisson

File: hack/generate-manifests.sh (hunk 2 above)
Comment (QuentinBisson) [reply hack/generate-manifests.sh:48]: They are here: https://github.com/kubernetes/csi-api/tree/master/pkg/apis/csi/v1alpha1
Can we use this path directly?

**Comment by tfussell** [comment]
@MarcelMue want to close this? #712 included the same commits.

**Pull request closed without merging**